arangod/Aql/Query.cpp

```diff
@@ -617,7 +616,10 @@ ExecutionState Query::execute(QueryResult& queryResult) {
         }
         // will set warnings, stats, profile and cleanup plan and engine
         auto state = finalize(*queryResult.extra);
-        if (state == ExecutionState::DONE && _cacheEntry != nullptr) {
+        bool isTransactionDisallowed = _transactionContext->isStreaming() &&
+                                       !_trx->state()->isReadOnlyTransaction();
+        if (state == ExecutionState::DONE && _cacheEntry != nullptr &&
+            !isTransactionDisallowed) {
```

Comment:
Can we change the naming here from `isTransactionDisallowed` to `isCachingDisallowed`, because that's what it actually is about. In fact, `isCachingAllowed` would be an ever better name, because then we can avoid the double negation in `! isTransactionDisallowed` (or `! isCachingDisallowed`).